Feat: Add Babylon phase 2 apis

oliviathet opened the pull request:
Features:
- add BBT5 Babylon network ID
- add slash deposit, slash early unbond, and withdraw slashing for Sign Babylon Staking API
- add Sign Babylon Registration API

Fixes/Chores:
- remove redundant strict decoder
- fix MFA Request Info `expiresAt`
- unmarshal Babylon Staking Request action
- strict unmarshal PSBT Sign Request
- removes duplicate Babylon Staking Params struct
- Babylon staking structs accepts array of finality provider pks instead of just one string

oliviathet committed:
disambiguate early unbond and withdrawal request in unmarshal json

oliviathet committed:
remove redundant decoder. fix mfa req expiresAt. unmarshal babylon staking req action. strict unmarshal psbt sign req

oliviathet committed:
remove another redundant decoder. add is empty struct to utils

oliviathet committed:
try psbt sign req unmarshal fix

oliviathet committed:
add babylon slash_deposit and slash_early_unbond actions and structs. add bbt5 babylon network id. fix babylon staking request unmarshal json

oliviathet committed:
add babylon slash_deposit and slash_early_unbond to client

hashxtree commented on api/v0/mfa_request_info.go:
```diff
@@ -3,7 +3,7 @@ package v0
 type MfaRequestInfo struct {
 	// DateTime measured in seconds since unix epoch. A wrapper type for serialization that encodes a `SystemTime` as a `u64` representing the number of seconds since `SystemTime::UNIX_EPOCH`.
-	ExpiresAt int64 `json:"timestamp"`
+	ExpiresAt int64 `json:"expires_at"`
```
I see `u64` in comment
```suggestion
	ExpiresAt uint64 `json:"expires_at"`
```

oliviathet committed:
Update api/v0/mfa_request_info.go

Co-authored-by: Peter <145032760+hashxtree@users.noreply.github.com>

oliviathet committed:
add new slashing types

oliviathet committed:
uint64 type for mfa expires at

oliviathet committed:
new btc staking params struct

oliviathet committed:
add back version

kwantam commented on api/v0/babylon_staking_request.go:
```diff
@@ -36,61 +39,77 @@ func BabylonStakingWithdrawalAsBabylonStakingRequest(v *BabylonStakingWithdrawal
+// BabylonStakingSlashDepositAsBabylonStakingRequest is a convenience function that returns BabylonStakingSlashDeposit wrapped in BabylonStakingRequest
+func BabylonStakingSlashDepositAsBabylonStakingRequest(v *BabylonStakingSlashDeposit) BabylonStakingRequest {
+	return BabylonStakingRequest{
+		BabylonStakingSlashDeposit: v,
```
I could be missing something, but doesn't this also need the `action` field to be set appropriately?

kwantam commented on api/v0/babylon_staking_request.go:
```diff
@@ -36,61 +39,77 @@ func BabylonStakingWithdrawalAsBabylonStakingRequest(v *BabylonStakingWithdrawal
+// BabylonStakingSlashEarlyUnbondAsBabylonStakingRequest is a convenience function that returns BabylonStakingSlashEarlyUnbond wrapped in BabylonStakingRequest
+func BabylonStakingSlashEarlyUnbondAsBabylonStakingRequest(v *BabylonStakingSlashEarlyUnbond) BabylonStakingRequest {
+	return BabylonStakingRequest{
+		BabylonStakingSlashEarlyUnbond: v,
```
ditto

kwantam commented on api/v0/babylon_staking_request.go:
```diff
@@ -3,14 +3,17 @@ package v0
+	BabylonStakingEarlyUnbond      *BabylonStakingEarlyUnbond
+	BabylonStakingWithdrawal       *BabylonStakingWithdrawal
+	BabylonStakingSlashDeposit     *BabylonStakingSlashDeposit
+	BabylonStakingSlashEarlyUnbond *BabylonStakingSlashEarlyUnbond
```
`BabylonStakingSlashDeposit` and `BabylonStakingSlashEarlyUnbond` are identical to `BabylonStakingEarlyUnbond`. You don't really need to create separate types, it'll just be a bunch of duplicate code.

kwantam commented on api/v0/babylon_staking_slash_deposit.go:
see comment above: this type is identical to `BabylonStakingEarlyUnbond`

kwantam commented on api/v0/babylon_staking_slash_early_unbond.go:
see comment above: this type is identical to `BabylonStakingEarlyUnbond`

kwantam reviewed (approved):
seems basically OK, but I think some parts could be simplified a bit

oliviathet committed:
remove duplicate babylon staking types

oliviathet committed:
add withdraw_slashing babylon staking type. add missing fields to babylon staking early unbond

oliviathet committed:
remove duplicate explicit staking params

oliviathet committed:
rename fp pk to fp pks

oliviathet committed:
use float64 when unmarshaling nums

oliviathet committed:
nit: capitalization

oliviathet committed:
nit: staking_params -> babylon_staking_params. add comment

oliviathet committed:
add as_base64 to babylon staking withdrawal. finality_provider_pk -> finality_provider_pks in serialization

oliviathet committed:
nit cleanup serialization for babylon structs

oliviathet commented:
@kwantam can you please take another look at it? 
- removed the dupe structs, now just using the existing `BabylonStakingEarlyUnbond` for slash_deposit and slash_early_unbond
- added withdraw_slashing, which uses the existing `BabylonStakingWithdrawal` struct
- changed the `finality_provider_pk` string param in the Babylon structs to a `finality_provider_pks` string array
- added `as_base64` to `BabylonStakingEarlyUnbond` and `BabylonStakingWithdrawal`
- removed a dupe struct for `BabylonStakingParams`

oliviathet committed:
add babylon registration api

oliviathet committed:
nit: comment

oliviathet committed:
use strings for some of the babylon staking params

oliviathet committed:
nit: rename PKs -> Pks. match staking param types with babylon rpc go sdk types

oliviathet committed:
add sign babylon registration scope

oliviathet committed:
try only fp pk single string for babylon deposit and registration

oliviathet committed:
in sign babylon deposit, change fp pk -> fp pks
